wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationEventProcessor.swift

```diff
@@ -150,14 +150,16 @@ public class ConversationEventProcessor: NSObject, ConversationEventProcessorPro
             from: event.payload
         ) else { return }
 
+        if let conversationID = payload.qualifiedID {
+            await conversationService.syncConversationIfMissing(qualifiedID: conversationID)
+        }
         await context.perform {
             self.processor.processPayload(
                 payload,
                 originalEvent: event,
                 in: self.context
             )
         }
-        await syncConversationForMLSStatus(payload: payload)
```

Comment:
@johnxnguyen idea what purpose this served? I don't see why we would need to update any MLS status upon receiving a `memberJoined` event, that's done when the welcome message arrives.

Reply:
IIRC This was done because the `memberJoin` and `welcome` events don't always arrive in the same order.
- `memberJoin` creates the `ZMConversation`  
- `welcome` creates the mls group

So when we receive a memberJoin event we need to set the correct `mlsStatus` depending on wether or not the mls group already exists.

Reply:
Ok I think this is not true anymore because inside `process(welcomeMessage:)` the first we do is to call

```
await conversationService.syncConversation(qualifiedID: conversationID)
```

I think this was logic was from the time when `conversationID` inside the WelcomeMessage event didn't reference the correct conversation.

See https://wearezeta.atlassian.net/browse/WPB-3435